### DISCUSSION TITLE
Some questions [beginner]

### DISCUSSION THREAD

**Discussion by zyt1024** [Xilinx/finn#841] [Q&A] [answered]
Hello, my model contains the `torch.topK` function. Does FINN support it?

**Comment by auphelia** [comment, marked as answer]
Hi @zyt1024 ,

We have a hardware implementation of the TopK node in FINN: [LabelSelect_Batch](https://github.com/Xilinx/finn/blob/main/src/finn/custom_op/fpgadataflow/labelselect_batch.py). Usually we don't export the model with the topk node from Brevitas directly, but we insert it after the export. You can have a look at the [tfc_end2end_example](https://github.com/Xilinx/finn/blob/main/notebooks/end2end_example/bnn-pynq/tfc_end2end_example.ipynb) to see how this is done, it is described in the _**Adding Pre- and Postprocessing**_ section.
